src/test/java/uk/gov/service/notify/ClientIntegrationTestIT.java

```diff
@@ -250,6 +250,32 @@ public void testGetTemplatePreview() throws NotificationClientException {
         assertTrue(template.getBody().contains(uniqueName));
     }
 
+    @Test
+    public void testGetReceivedTextMessages() throws NotificationClientException {
+        String apiKey = System.getenv("INBOUND_SMS_QUERY_KEY");
+        String baseUrl = System.getenv("NOTIFY_API_URL");
+        NotificationClient client = new NotificationClient(apiKey, baseUrl);
+
+        ReceivedTextMessageList response = client.getReceivedTextMessages(null);
+        assertFalse(response.getReceivedTextMessages().isEmpty());
+        assertNotNull(response.getCurrentPageLink());
+        ReceivedTextMessage receivedTextMessage = response.getReceivedTextMessages().get(0);
+        assertNotNull(receivedTextMessage.getId());
+        assertNotNull(receivedTextMessage.getNotifyNumber());
+        assertNotNull(receivedTextMessage.getUserNumber());
+        assertNotNull(receivedTextMessage.getContent());
+        assertNotNull(receivedTextMessage.getCreatedAt());
+        assertNotNull(receivedTextMessage.getServiceId());
+
+        testGetReceivedTextMessagesWithOlderThanId(receivedTextMessage.getId(), client, response.getReceivedTextMessages().size());
+    }
+
+    private void testGetReceivedTextMessagesWithOlderThanId(UUID id, NotificationClient client, int expectedCount) throws NotificationClientException {
+        ReceivedTextMessageList response = client.getReceivedTextMessages(id.toString());
+        assertNotNull(response.getReceivedTextMessages());
+        assertEquals(expectedCount-1, response.getReceivedTextMessages().size());
```

Comment:
Not sure if `assertEquals(expectedCount-1, response.getReceivedTextMessages().size());` is useful as the `expectedCount` is being passed in by the same method, and there is a very slim chance that another client could send a message to the inbound sms endpoint failing the test.

Also should the Java client have the sms_sender test updated to use the `API_SENDING_KEY` in line with other clients? if so will also need to have the `SMS_SENDER_ID` updated to use a valid number rather than `fromCIT` sender in the AWS repo.

Reply:
The purpose is to test that I should get one less in the list that the previous call without the olderThan parameter. This should work even if another message is added because it is only return messages older than that id. Newer messages will not be included in the list.

Reply:
Why do we need the API_SENDING_KEY? Is that extra?

Reply:
`API_SENDING_KEY` is a whitelist API key for the inbound sms, there should however be texts sent out from the python and Net integration tests so unlikely to not be able to retrieve some received text messages.